contracts/Gate.sol

```diff
@@ -38,9 +50,117 @@ contract Gate is ProtocolVersioned {
      *  Storage
      */
     address public workers; //TODO: update type once workers contract and interfaces are added to the repo.
+
+    // stake requests
     mapping(address /*staker */ => StakeRequest) public stakeRequests;
 
-    function Gate() public { //TODO: complete as necessary
+    // bounty amount
+    uint256 private bounty;
+
+    // utility token UUID
+    bytes32 private uuid;
+
+    // OpenSTValue contract
+    OpenSTValue private openSTValue;
+
+    /*
+     *  Public functions
+     */
+    constructor(
+        address _workers,
+        uint256 _bounty,
+        bytes32 _uuid,
+        address _openSTValue)
+        public
+        ProtocolVersioned(_openSTValue)
+    {
+        require(_workers != address(0));
+        require(_openSTValue != address(0));
+        require(_uuid.length != 0);
+
+        // Should bounty with 0 amount be allowed ?
+        //require(_bounty > 0);
+
+        workers = _workers;
+        bounty = _bounty;
+        uuid = _uuid;
+        openSTValue = OpenSTValue(_openSTValue);
+    }
+
+    function requestStake(
+        uint256 _amount,
+        address _beneficiary)
+        external
+        returns ( bool /* success */)
+    {
+
+        require(_amount > 0);
+        require(_beneficiary != address(0));
+
+        // check if the state request does not exists
+        require(address(stakeRequests[msg.sender].beneficiary) == address(0));
+
+        require(openSTValue.valueToken().allowance(msg.sender, address(this)) >= _amount);
+        require(openSTValue.valueToken().transferFrom(msg.sender, address(this), _amount));
+
+        stakeRequests[msg.sender] = StakeRequest({
+            amount: _amount,
+            beneficiary: _beneficiary,
+            hashLock: 0,
+            unlockHeight: 0
+        });
 
+        emit StakeRequested(msg.sender, _amount, _beneficiary);
+
+        return true;
+    }
+
+    function revertStakeRequest()
+        external
+        returns (bool /* success */)
+    {
+        StakeRequest storage stakeRequest = stakeRequests[msg.sender];
+
+        // check if the state request exists for the msg.sender
+        require(address(stakeRequest.beneficiary) != address(0));
+
+        // check if the stake request was not accepted
+        require(stakeRequest.unlockHeight == 0);
+
+        require(openSTValue.valueToken().transfer(msg.sender, stakeRequest.amount));
+
+        delete stakeRequests[msg.sender];
+
+        emit StakeRequestReverted(msg.sender, stakeRequest.amount);
```

Comment:
would stakeRequest.amount would be zero after its deleted from the mapping? This question also applies to the other delete in the reject function.

Reply:
Yes, stakeRequest.amount would be 0

Reply:
Ahh, I overlooked the comment before. Yes it should return the actual value. Made changes